chef/cookbooks/hyperv/recipes/do_ceilometer.rb

```diff
@@ -0,0 +1,8 @@
+raise if not node[:platform] == 'windows'
+
+include_recipe "hyperv::openstack_install_ceilometer"
+include_recipe "hyperv::openstack_install_neutron"
+include_recipe "hyperv::config_neutron"
+include_recipe "hyperv::config_ceilometer"
+include_recipe "hyperv::register_services_ceilometer"
+include_recipe "hyperv::register_services_neutron"
```

Comment:
True. But there are some bits that IMHO makes this approach better:

* The `role_constrains` is decoupled.  If we allows some new agent to be installed in a Windows machine, we need to rewrite this part to separate the code as is done now.
* The barclamp-ceilometer allows you to install the agent independently from what is installed in the node.  This is true for GNU/Linux too
* The barclamp-ceilometer suggest automatically the correct Hyper-V node when you want to deploy Ceilometer in the cloud.

So, basically, if we constrain the installation of Ceilometer in HyperV nodes only when you have deployed Nova, we behave differently for the rest of the barclamp-ceilometer.